enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,609 @@
+> This behavior may require explicit documentation, or expanding the phase 0 design scope to be able to accommodate all the catalog sources in the cluster.
+
+After determining the set of bundles that need to exist in order to satisfy admin package intent, the PO manager's installation layer will need to attempt to install those bundles. The management of those bundle contents will be delegated to OLM's rukpak layer, and the POM component would be responsible for generating an underlying BundleDeployment resource for the bundle that was selected.
+This underlying, cluster-scoped BundleDeployment resource will contain owner references to the parent PlatformOperator custom resource the manager is reconciling.
+
+The POM component will also be responsible for bubbling up any failure states during reconciliation:
+
+- Failure to successfully source a bundle from the admin configured OLM package name.
+- Failure to successfully install the OLM-based operator after determining a bundle that meets the desired package name intent.
+
+In both cases, the POM component will relay those failure states back to the PlatformOperator custom resource's status sub-resource that it's currently reconciling. Any PlatformOperator custom resources that are expressing failing status condition types will eventually be reflected in the "aggregate" ClusterOperator resource.
+
+Admins may elect to install or delete a PO package after cluster creation. This process would include creating, or deleting, a PlatformOperator custom resource. In order to delete a PO after cluster creation, admins would need to delete the top-level PlatformOperator custom resource for that package.
+
+**Note**: Deleting the top-level PlatformOperator custom resource will result in the cascading deletion of bundle contents. This may result in Kubernetes garbage collecting cluster-scoped resources, e.g. CRDs, which can impact workload health in unsupported configurations.
+This behavior differs from the current OLM 0.x behavior, which has an opinionated tenancy model it needs to protect, and can lead to data loss scenarios.
+
+In the case that a bundle version has been rolled out to the cluster that's vulnerable, or broken in some capacity, admins may perform the following manual steps to upgrade the PO package version:
+
+- Find the current bundle version for the configured PO package name.
+- Verifying that the redhat-operators index image contains a newer, higher semver bundle than what's currently applied to the cluster.
+- Deleting the underlying bundle deployment resources to force the PO manager to select a new bundle.
+
+These steps may result in workload disruption while new bundle contents are rolled out on the cluster and Kubernetes garbage collects the old bundle contents, and this behavior may require explicit documentation.
```

Comment:
```suggestion
These steps will very likely result in workload disruption while new bundle contents are rolled out on the cluster and Kubernetes garbage collects the old bundle contents, and this behavior may require explicit documentation.
```

Any APIs defined by that PO will disappear, which means admins will have to re-create CRs after the delete/create cycle for that PO completes.

Reply:
Updated in https://github.com/openshift/enhancements/pull/1170/commits/bc14b4537125d43153cc452cd686e02cf96630b2.